cgp/genome.py

```diff
@@ -90,6 +90,9 @@ def __init__(
         # dictionary to store values of Parameter nodes
         self._parameter_names_to_values: Dict[str, float] = {}
 
+        # list of permissible values for every gene
+        self._permissible_values: List[np.ndarray[int]] = self.determine_permissible_values()
```

Comment:
I think the `np.ndarray[int]` is not a valid type hint. However, `mypy` anyway ignores all numpy type hints, because numpy isn't type annotated, so this does not lead to a problem.

To be consistent with the other hints, I'd recommend to remove the `[int]` here.